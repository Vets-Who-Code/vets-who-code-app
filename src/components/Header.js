import React, { Component } from 'react';

// import HeaderSlider from './Slider'

import Slider from "react-slick";
import Link from 'gatsby-link'


import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css'


import codePNG from '../images/code.png'
import speakingPNG from '../images/speaking.png'
import thisIsUsPNG from '../images/this_is_us.png'

class Header extends Component {
  constructor(props) {
    super(props);
    this.play = this.play.bind(this);
    this.next = this.next.bind(this);
    this.previous = this.previous.bind(this);
  }

  play() {
    this.slider.slickPlay();
  }

  next() {
    this.slider.slickNext();
  }

  previous() {
    this.slider.slickPrev();
  }

  render() {
    const settings = {
      dots: true,
      infinite: true,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
      autoplay: true,
      autoplaySpeed: 3000,
    }
    return (
      <header id="site-header" className="site-header flexslider classic">
        <Slider {...settings} ref={slider => this.slider = slider }>
        <div>
          <div style={{ backgroundImage: `url(${codePNG})`, backgroundSize: 'cover'}}>
            <div className="header-classic  wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-left">
                      <h1>Learn</h1>
                      <p className="subtitle">How To Code With Other Veterans.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/apply" className="btn btn-charity-default">
                          Apply
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div style={{ backgroundImage: `url(${speakingPNG})`, backgroundSize: 'cover'}}>
            <div className="header-classic  wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-left">
                      <h1>Sign Up</h1>
                      <p className="subtitle">As A Mentor.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/mentor" className="btn btn-charity-default">
                          {' '}
                          Sign Up
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div style={{ backgroundImage: `url(${thisIsUsPNG})`, backgroundSize: 'cover'}}>
            <div className="header-classic wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-right">
                      <h1 className="red">
                        <span className="red">Help Us</span>
                      </h1>
                      <p className="subtitle red">
                        Teach More Veterans How To Code.
                      </p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/donate" className="btn btn-charity-default">
                          Please Donate
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </Slider>
      <ul className="flex-direction-nav" onClick={this.previous}>
        <li className="flex-nav-prev">
          <a className="flex-prev" href="#">
            Previous
          </a>
        </li>
        <li className="flex-nav-next" onClick={this.next}>
          <a className="flex-next" href="#">
            Next
          </a>
        </li>
      </ul>
    </header>
    );
  }
}

// const Header = () => {
//   return (


//   )
// }

export default Header