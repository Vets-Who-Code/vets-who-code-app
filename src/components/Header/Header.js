import Link from 'next/link'
import Image from 'next/image'
import Typed from 'react-typed'

function Header() {
  return (
    <section className="site-header flexslider classic overlay main-overlay grey">
      <Image
        layout="fill"
        blurDataURL="/images/code.jpg"
        placeholder="blur"
        src="/images/code.jpg"
        alt="image of code on a computer screen"
      />
      <div className="header-classic wrapper-table">
        <div className="valign-center">
          <div className="container">
            <div className="col-md-10 col-md-offset-1">
              <div className="intro text-left" style={{ color: '#fff' }}>
                <h1>
                  <Typed
                    className="typedString"
                    strings={[
                      'Learn How To Code',
                      'Build Open Source',
                      'Network With Peers',
                      'Get A Job',
                    ]}
                    typeSpeed={70}
                    backSpeed={80}
                    smartBackspace
                    loop
                  />
                </h1>
                <p className="subtitle">With Vets Who Code.</p>
                <div className="btn-cal-group">
                  <Link href="/apply" className="btn btn-charity-default">
                    <a className="btn btn-charity-default">Apply</a>
                  </Link>
                  &nbsp;
                  <Link href="/donate">
                    <a className="btn btn-charity-default">Donate</a>
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  )
}

export default Header
