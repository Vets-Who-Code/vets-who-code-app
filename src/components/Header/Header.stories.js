import React from 'react'
import ThemeProvider from '../../store/ThemeProvider'
import Header from './Header'

export default {
  component: Header,
  title: 'Components/Header',
}

const Template = args => <Header {...args} />

export const Default = Template.bind({})

Default.args = {}
