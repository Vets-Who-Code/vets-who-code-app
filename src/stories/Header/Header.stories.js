import Header from '../../../src/components/Header'

export default {
  component: Header,
  title: 'Components/Header',
}

const Template = args => <Header {...args} />

export const Example = Template.bind({})

Example.args = {}
