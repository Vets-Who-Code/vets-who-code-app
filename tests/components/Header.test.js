import { render } from '@testing-library/react'
import Header from '@/components/Header'

describe('<Header />', () => {
  test('should render correctly', () => {
    const { container } = render(<Header />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
