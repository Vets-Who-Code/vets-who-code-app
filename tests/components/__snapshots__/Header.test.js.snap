// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<section
  class="site-header flexslider classic overlay main-overlay grey"
>
  <img
    alt="image of code on a computer screen"
    layout="fill"
    placeholder="blur"
    src="/images/code.jpg"
  />
  <div
    class="header-classic wrapper-table"
  >
    <div
      class="valign-center"
    >
      <div
        class="container"
      >
        <div
          class="col-md-10 col-md-offset-1"
        >
          <div
            class="intro text-left"
            style="color: rgb(255, 255, 255);"
          >
            <h1>
              <span
                class="typedString"
              >
                <span />
                <span
                  class="typed-cursor"
                >
                  |
                </span>
              </span>
            </h1>
            <p
              class="subtitle"
            >
              With Vets Who Code.
            </p>
            <div
              class="btn-cal-group"
            >
              <a
                class="btn btn-charity-default"
                href="/apply"
              >
                Apply
              </a>
              Â 
              <a
                class="btn btn-charity-default"
                href="/donate"
              >
                Donate
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
